### PR TITLE
Remove leaking internals from libchirp include headers

### PR DESCRIPTION
**Pull request by ganwell**
Byproducts:

* Fix recursion error in Makefile generation
* Ignore floobits files

Protip: I renamed include/libchirp/common.h to src/common.h and created a new include/libchirp/common.h the diff is actually quite small. Just check include/libchirp/common.h and check that there is no duplicated code in src/common.h

### CODE REVIEW

**Comment by ganwell** [comment]
@sliverc See comments about diff in description

File: src/common.h [hunk 1]
```diff
@@ -0,0 +1,428 @@
...
+                chirp->_log(buf, 1); \
+            } else { \
+                fprintf( \
+                    stderr, \
+                    "%s:%d " message "\n", \
+                    __FILE__, \
+                    __LINE__, \
+                    __VA_ARGS__ \
+                ); \
+                assert(condition); \
```
Comment (sliverc) [inline src/common.h]: what is the reasons that double assert here... maybe need to elaborate a bit further what could make condition unstable?

Comment (ganwell) [inline src/common.h]: @sliverc I wrote a chapter about double evaluation, please check it.

File: src/common.h [hunk 2]
```diff
@@ -0,0 +1,477 @@
...
+//
+
+#include <assert.h>
+#include <limits.h>
+#include <string.h>
+#include <stdlib.h>
+#include <stddef.h>
+#include <stdio.h>
+
+// Logging and assert macros
```
Comment (ganwell) [inline src/common.h]: @sliverc Here is it.

File: src/common.h (hunk 1 above)
Comment (sliverc) [inline src/common.h]: nice this makes it clear

**Pull request merged**